### PR TITLE
SWC-6776 - Allow updating ACL immediately after upload

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx [hunk 1]
```diff
@@ -507,58 +518,6 @@ describe('EntityAclEditor', () => {
     ).toBeInTheDocument()
   })
 
-  it('displays an error on mutate failure', async () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx:510]: moved this test to the end, without changes.

This test dynamically updates a handler to return an error, but for some reason `server.restoreHandlers` didn't fix it. Not sure why, but moving the test fixes the issue

File: packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx [hunk 2]
```diff
@@ -281,7 +292,7 @@ describe('EntityAclEditor', () => {
       },
       mockProject.bundle.benefactorAcl.resourceAccess.length,
     )
-    verifyHasLocalSharingSettingsMessage()
+    verifyHasProjectMessage()
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx:295]: Fixed a subtle bug where this scenario showed the wrong copy.

File: packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx [hunk 3]
```diff
@@ -626,4 +585,97 @@ describe('EntityAclEditor', () => {
...
+          ...mockProject.bundle.accessControlList,
+          resourceAccess: expect.any(Array),
+        },
+        MOCK_ACCESS_TOKEN,
+      )
+      expect(onUpdateSuccess).toHaveBeenCalled()
+      expect(sendMessageSpy).not.toHaveBeenCalled()
+    })
+  })
+  it('displays an error on mutate failure', async () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx:630]: test moved from above without changes

File: packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx [hunk 4]
```diff
@@ -626,4 +585,97 @@ describe('EntityAclEditor', () => {
       'You do not have sufficient privileges to modify the sharing settings.',
     )
   })
+
+  it('shows appropriate UI in the post-upload scenario', async () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.test.tsx:589]: test new functionality

File: packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.tsx [hunk 5]
```diff
@@ -318,7 +334,38 @@ const EntityAclEditor = React.forwardRef(function EntityAclEditor(
...
+          isGlobal={false}
+          variant={'warning'}
+          title={`Edits will affect settings of entire ${entityTypeToFriendlyName(
+            entityBundle?.entityType,
+          ).toLowerCase()}.`}
+          description={
+            <>
+              <p>
+                Editing the settings here will impact the sharing settings for
+                all files and folders within this project, not just the ones
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.tsx:350]: Should "within this project" be "within this ${entityTypeToFriendlyName(entityBundle?.entityType).toLowerCase()"

Comment (nickgros) [reply packages/synapse-react-client/src/components/EntityAclEditor/EntityAclEditor.tsx:350]: yep, thanks for catching that!

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**